src/lexers/jsx-lexer.js

```diff
@@ -1,120 +1,69 @@
-import * as acorn from 'acorn-jsx'
+import * as walk from 'acorn/dist/walk'
 import assert from 'assert'
-import HTMLLexer from './html-lexer'
-import BaseLexer from './base-lexer'
 import { ParsingError } from '../helpers'
 
-export default class JsxLexer extends HTMLLexer {
-  constructor(options = {}) {
-    options.attr = options.attr || 'i18nKey'
-    super(options)
-  }
-
-  extract(content) {
-    this.extractInterpolate(content)
-    this.extractTrans(content)
-    return this.keys
-  }
-
-  extractInterpolate(content) {
-    let matches
-    const regex = new RegExp(
-      '<Interpolate([^>]*\\s' + this.attr + '[^>]*)\\/?>',
-      'gi'
-    )
-
-    while (matches = regex.exec(content)) {
-      const attrs = this.parseAttributes(matches[1])
-      const key = attrs.keys
-      if (key) {
-        this.keys.push({ ...attrs.options, key })
+export const JSXParserExtension = Object.assign({}, walk.base, {
+  JSXText(node, st, c) {
+    // We need this catch, but we don't need the catch to do anything.
+  },
+  JSXElement(node, st, c) {
+      node.openingElement.attributes.forEach((attr) => {
+          c(attr, st, attr.type);
+      });
+      node.children.forEach((child) => {
+          c(child, st, child.type);
+      });
+  },
+  JSXExpressionContainer(node, st, c) {
+      c(node.expression, st, node.expression.type);
+  },
+  JSXAttribute(node, st, c) {
+      if (node.value !== null) {
+          c(node.value, st, node.value.type);
       }
-    }
-
-    return this.keys
+  },
+  JSXSpreadAttribute(node, st, c) {
+      c(node.argument, st, node.argument.type);
   }
+});
 
-  /**
-  * Extract tags and content from the Trans component.
-  * @param {string} string
-  * @returns {array} Array of key options
-  */
-  extractTrans(content) {
-    let matches
-    const selfClosingTagPattern = '(?:<\\s*Trans([^>]*)?/>)'
-    const closingTagPattern = '(?:<\\s*Trans([^>]*)?>((?:(?!</\\s*Trans\\s*>)[^])*)</\\s*Trans\\s*>)'
-    const regex = new RegExp(
-      [selfClosingTagPattern, closingTagPattern].join('|'),
-      'gi'
-    )
+export function nodeToString(ast, string) {
```

Comment:
This is identical to the `extractTrans` method except it takes an already-parsed node and not a string. Moved out of the class.